examples/shell/shell_common/cmd_otcli.cpp

```diff
@@ -40,11 +40,13 @@
 #include <openthread/link.h>
 #include <openthread/thread.h>
 #if OPENTHREAD_API_VERSION >= 85
+#if defined(__ZEPHYR__)
```

Comment:
We shouldn't be adding platform specific defines like this in these platform-agnostic files. Why does Zephyr specifically need this treatment?

Reply:
For at least our platform, the ot-ctli needs to be enabled at the same time as the openthread task for it not to cause problemes with the commiossioning.

I added the `defined(__ZEPHYR__)` because nordic where the ones that had implemented the initial version of this code. The goal was to not break their platform and get ours to work.

Side not, where we enable the ot-cli in the openthread init, there is define for it not to be included on zephyr platforms.

Reply:
The only solution I can think of is having define that determines which init to use for the ot-cli. @mrjerryjohns, what do you think?

Reply:
Looking at this, i cannot agreed more with @mrjerryjohns 
You should sync with Nordic to know why it is working for them and not for SiLabs.
This is clearly a workaround not solving the root cause.
The problem might be located directly with the way openthread is included and configured for EFR32 freertos compare to NRF zephyr - IMO we should not block the PR but a TODO / issue must be added for further investigation

Reply:
this is odd indeed ... !

Reply:
In that case, can you please normalize this to add a define like `ENABLE_OT_CLI` (or something equivalent) that avoids the mentioning of ZEPHYR, and on the Zephr platform config, enable/disable those defines as needed.

Reply:
Changed the zephyr define with `CHIP_DEVICE_CONFIG_THREAD_ENABLE_CLI` which is by default 0 so it will not affect zephyr platforms.
Platforms already using the define will not be affected by the use of the define.